### PR TITLE
HDDS-11082. Code cleanup in the DatanodeStateMachine class

### PR DESCRIPTION
**Pull request by jianghuazhu**
## What changes were proposed in this pull request?
There are some redundant codes in DatanodeStateMachine, and the purpose of this PR is to optimize them.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-11082

## How was this patch tested?
We need to ensure that the unit tests pass normally.

### CODE REVIEW

**Comment by jianghuazhu** [comment]
ci : https://github.com/jianghuazhu/ozone/actions/runs/9711680423
@adoroszlai @hemantk-12 @errose28 , can you help review this PR.
thank you.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeStateMachine.java [hunk 1]
```diff
@@ -281,8 +281,7 @@ public DatanodeStateMachine(DatanodeDetails datanodeDetails,
   private int getEndPointTaskThreadPoolSize() {
     // TODO(runzhiwang): current only support one recon, if support multiple
     //  recon in future reconServerCount should be the real number of recon
-    int reconServerCount = 1;
-    int totalServerCount = reconServerCount;
```
Comment (adoroszlai) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeStateMachine.java:285]: The comment mentions `reconServerCount`.  If you remove that variable, please adjust the comment accordingly.

**Review by adoroszlai — COMMENTED** [review]
Thanks @jianghuazhu for the cleanup.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeStateMachine.java [hunk 2]
```diff
@@ -625,9 +621,9 @@ public synchronized void stopDaemon() {
       context.setState(DatanodeStates.SHUTDOWN);
       reportManager.shutdown();
       this.close();
-      LOG.info("Ozone container server stopped.");
+      LOG.info("The DatanodeStateMachine stopped.");
     } catch (IOException e) {
-      LOG.error("Stop ozone container server failed.", e);
+      LOG.error("Stop DatanodeStateMachine failed.", e);
```
[the remark this answers is not in the document]
Comment (jianghuazhu) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeStateMachine.java:626]: Thanks @adoroszlai  for the comments and review.
I'll update later.

**Comment by jianghuazhu** [comment]
I've updated some code.
An error was found in the CI, I tried to execute it several times, but it still couldn't pass.
Could you please take a look, @adoroszlai .
thank you.
https://github.com/jianghuazhu/ozone/actions/runs/9891414676/job/27362321602

**Comment by adoroszlai** [comment]
@jianghuazhu it can be fixed by merging `origin/master` into your branch

**Comment by jianghuazhu** [comment]
I have updated it and CI has passed.
Could you help review again, @adoroszlai .
thanks.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeStateMachine.java [hunk 3]
```diff
@@ -701,15 +697,6 @@ private Thread getCommandHandlerThread(Runnable processCommandQueue) {
     return handlerThread;
   }
 
-  /**
-   * Returns the number of commands handled  by the datanode.
-   * @return  count
-   */
-  @VisibleForTesting
-  public long getCommandHandled() {
```
Comment (fapifta) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeStateMachine.java:709]: Once this method is removed, the commandsHandled long value is just incremented in the code, but it is never read, can we remove that as well?

**Review by fapifta — COMMENTED** [review]
Thank you @jianghuazhu for your work on this one, please find one minor comment, other than that, it looks good to me.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeStateMachine.java (hunk 3 above)
Comment (jianghuazhu) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeStateMachine.java]: Thanks for your comment and review, @fapifta .
I have updated it.

Comment (fapifta) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeStateMachine.java]: Hi @jianghuazhu, thank you for the update, I did not meant to re-add the method again. If the method is not used anymore, let's remove it, but remove it together with the commandsHandled variable, as that does not have meaningful uses if this getter is not used.

Comment (jianghuazhu) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeStateMachine.java:709]: Thanks @fapifta  for the comment and review.
I updated it again.

**Review by fapifta — APPROVED** [review]
Thank you @jianghuazhu for your continued efforts, +1, pending CI

**Comment by jianghuazhu** [comment]
There is a CI that has failed, we can try to restart it, @fapifta .

**Comment by adoroszlai** [comment]
Thanks @jianghuazhu for the patch, @fapifta for the review.

**Pull request merged**